src/diffusers/pipelines/controlnet/pipeline_controlnet.py

```diff
@@ -976,8 +977,17 @@ def __call__(
 
         # 8. Denoising loop
         num_warmup_steps = len(timesteps) - num_inference_steps * self.scheduler.order
+        is_unet_compiled = is_compiled_module(self.unet)
+        is_controlnet_compiled = is_compiled_module(self.controlnet)
+        is_torch_higher_equal_than_2_1 = version.parse(version.parse(torch.__version__).base_version) >= version.parse(
+            "2.1"
+        )
         with self.progress_bar(total=num_inference_steps) as progress_bar:
             for i, t in enumerate(timesteps):
+                # Relevant thread:
+                # https://dev-discuss.pytorch.org/t/cudagraphs-in-pytorch-2-0/1428
+                if (is_unet_compiled and is_controlnet_compiled) and is_torch_higher_equal_than_2_1:
+                    torch._inductor.cudagraph_mark_step_begin()
```

Comment:
Please check:
https://huggingface.slack.com/archives/C047QHKH944/p1697736070025029

Reply:
Hmm this is in some sense a breaking change from PT, do we really have to add version specific code here?

Reply:
There is no other way to support compiled ControlNets otherwise in PT 2.1, sadly.